doc/sphinx/proof-engine/tactics.rst

```diff
@@ -105,6 +105,227 @@ bindings_list`` where ``bindings_list`` may be of two different forms:
   .. exn:: Not the right number of missing arguments.
      :undocumented:
 
+.. _intropatterns:
+
+Intro patterns
+~~~~~~~~~~~~~~
+
+Intro patterns let you specify the name to assign to variables and hypotheses
+introduced by tactics.  They also let you split an introduced hypothesis into
+multiple hypotheses or subgoals.  Common tactics that accept intro patterns
+include :tacn:`intros`, :tacn:`destruct` and :tacn:`apply`.
+
+.. productionlist::        coq
+   simple_intropattern     : `simple_intropattern_closed`
+                           : % `operconstr` ... % `operconstr`
+   simple_intropattern_closed : `or_and_intropattern`
+                              : `equality_intropattern`
+                              : _
+                              : `naming_intropattern`
+   or_and_intropattern     : [ `intropatterns` | ... | `intropatterns` ]
+                           : ()
+                           : ( `simple_intropattern` , ... , `simple_intropattern` )
+                           : ( `simple_intropattern` & ... & `simple_intropattern` )
+   equality_intropattern   : ->
+                           : <-
+                           : [= `intropatterns` ]
+   intropatterns           : `nonsimple_intropattern` ...
+                           : <empty>
```

Comment:
It represents the empty production.  Otherwise it would just be an empty line, which could be interpreted as just a blank line.

![image](https://user-images.githubusercontent.com/1253341/50367732-0d8a0500-0537-11e9-90b7-3683c664df9c.png)